clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -396,26 +421,55 @@ private Result checkIfNodesAreUpOrRetired(ClusterState clusterState) {
     }
 
     private Result checkRedundancy(DistributorNodeInfo distributorNodeInfo, Node node) {
-        List<StorageNode> storageNodes = distributorNodeInfo.getHostInfo().getDistributor().getStorageNodes();
-        for (StorageNode storageNode : storageNodes) {
-            if (storageNode.getIndex() == node.getIndex()) {
-                Integer minReplication = storageNode.getMinCurrentReplicationFactorOrNull();
-                // Why test on != null? Missing min-replication is OK (indicate empty/few buckets on system).
-                if (minReplication != null && minReplication < requiredRedundancy) {
-                    return disallow("Distributor " + distributorNodeInfo.getNodeIndex()
-                            + " says storage node " + node.getIndex()
-                            + " has buckets with redundancy as low as "
-                            + storageNode.getMinCurrentReplicationFactorOrNull()
-                            + ", but we require at least " + requiredRedundancy);
-                } else {
-                    return allow();
-                }
+        Integer minReplication = minReplication(distributorNodeInfo).get(node.getIndex());
+        return verifyRedundancy(distributorNodeInfo, minReplication, node.getIndex());
+    }
+
+    private Result checkRedundancySeenFromDistributor(DistributorNodeInfo distributorNodeInfo, Set<Integer> indexesToCheck) {
+        Map<Integer, Integer> replication = new LinkedHashMap<>(minReplication(distributorNodeInfo));
```

Comment:
Do we need to copy the returned map here?

Reply:
No, can't see why that's needed